lib/use_cases/automate_sending_letters.rb

```diff
@@ -0,0 +1,47 @@
+module UseCases
+  class AutomateSendingLetters
+    def initialize(case_ready_for_automation:, case_classification_to_letter_type_map:, generate_and_store_letter:, send_income_collection_letter:)
+      @case_ready_for_automation = case_ready_for_automation
+      @case_classification_to_letter_type_map = case_classification_to_letter_type_map
+      @generate_and_store_letter = generate_and_store_letter
+      @send_income_collection_letter = send_income_collection_letter
+    end
+
+    def execute(case_priority:)
+      automate_letters(case_priority: case_priority) if enviornment_allow_to_send_automated_letters?
+    end
+
+    private
+
+    def automate_letters(case_priority:)
+      return false unless @case_ready_for_automation.execute(patch_code: case_priority.patch_code)
+
+      income_collection_letters = %w[income_collection_letter_1 income_collection_letter_2]
+
+      letter_name = @case_classification_to_letter_type_map.execute(case_priority: case_priority)
+
+      return nil unless income_collection_letters.include?(letter_name)
```

Comment:
Probably only want to return `true` or `false` from this method, rather than `nil` too.
```suggestion
      return false unless income_collection_letters.include?(letter_name)
```